### ISSUE TITLE
Developing eval_utils module 

### ISSUE THREAD

**Issue by jarumihooi** [clamsproject/aapb-evaluations#42]
### Because

As a reformat task, and for future development of more evaluations, it should be considered if a common evaluation utilities module should be created.
- [ ] where should that module reside? Should it be a subsection of the clams_utils?

That module could contain commonly used code such as: 
- goldretriever
- data extraction/preprocessing from golds and preds
- guid/file matching between golds and preds. 
- common accuracy metrics
- etc

Note, not every evaluation will be similar, and thus some may use different metrics. As such, its likely that printing is unlikely to be universal enough for modulization. 

The environment for CLAMS so far seems to require: 
- ? [clams-python](https://github.com/clamsproject/clams-python)
- [mmif-python](https://github.com/clamsproject/mmif-python)
- [clams_utils](https://github.com/clamsproject/clams-utils) for goldretriever, which is used in each eval, and in the annotation for NEL. The goldretriever code seems to be commonly used. 
Is it better to place certain codes in utilities? What is the organizational method?

**Comment by keighrim** [comment]
Also related to #41, I think the most straightforward way of doing this will be providing a PyPI distro (`clams-aapb-eval` package hereinafter, working name for now) that includes all the utility modules and `Evaluator` superclass (ABC), so that evaluator developers can just `pip-install` and start writing. Then, conveniently, since I already set up [`clams-utils`](https://github.com/clamsproject/clams-utils) repo for daily PyPI release, I'm very tempted to use that to hold an additional `clams-aapb-eval` package . But there could be some issues with that approach, namely; 

1. there's no inherit linkage between this repo and `clams-utils` repo, so we need a very clear documentation for this arbitrary use of separate repositories. As I stated also in https://github.com/clamsproject/clams-python/issues/147, I don't like this arbitrariness / segregation of clearly related components. 
2. I'm not seeing a big chance that parts of `clams-aapb-eval` package is _commonly_ used in other part of the whole clamsproject. Hence ,no need to "factorize" the eval package. 
3. `clams-utils` is auto-released every day. If there's any problem in, for example, the `clams.utils.aapb.eval` package, even if it's a very minor bug, it will take time to do the fix and propagate the fix over PyPI. 

Combining the above reasons, I firmly believe the package should reside in this repository. This will imply 

1. we put a `aapb_eval` package in the project root
2. all other evaluator subdirectories should now be python packages (`__init__.py`)
3. all invocations of the `evalute.py`s should now through `python -m subdir.evaluate <more> <flags> <as> <defined>`
4. when evaluator developer found an issue with the superclass, they can directly mend and push the fix

**Comment by jarumihooi** [comment]
Conversation of where this repo is to be placed had two different outcomes,
can the comments above this be reviewed and updated plus any other details
of this development?

**Comment by keighrim** [comment]
> Conversation of where this repo is to be placed had two different outcomes,

I don't think I follow this. This repo doesn't go anywhere else...?

**Comment by keighrim** [comment]
From my comment above, 

> 1. we put a aapb_eval package in the project root
> 1. all other evaluator subdirectories should now be python packages (__init__.py)
> 1. all invocations of the evalute.pys should now through python -m subdir.evaluate <more> <flags> <as> <defined>
> 1. when evaluator developer finds an issue with the superclass, they can directly mend and push the fix

- number 1 is implemented in #72 
- number 2 and 3 should be part of #41 and will be addressed in a separate PR 
- number 4 is true after #72

**Comment by keighrim** [comment]
What #72 does not fully implement is scaffolding/templating of a new evaluation task, which started in #46 . There is a dedicated issue to discuss the topic (#37), so I hope we will have a new PR to address that issue in the future.
